packages/reaction-accounts/server/methods/accounts.js

```diff
@@ -127,20 +127,33 @@ Meteor.methods({
     return false;
   },
 
-  /*
-   * add new addresses to an account
+  /**
+   * accounts/addressBookAdd
+   * @description add new addresses to an account
+   * @param {Object} doc - address
+   * @param {String} accountId - `account._id` which is need to be updated
+   * @return {Object} with keys `numberAffected` and `insertedId` if doc was
+   * inserted
    */
   "accounts/addressBookAdd": function (doc, accountId) {
     check(doc, ReactionCore.Schemas.Address);
     check(accountId, String);
+    if (accountId !== Meteor.userId() && !ReactionCore.hasAdminAccess()) {
```

Comment:
The publication should take care of security for user accounts, as normal users are only subscribed to their own account data. The super admin gets access to everything, while shop admins can only access members of their shop.

https://github.com/reactioncommerce/reaction/blob/development/packages/reaction-accounts/server/publications/accounts.js

Doing the permission check here might be redundant, and might affect what's happening with the publication. We should make sure the publication is indeed preventing unwanted access to accounts; Then make sure these methods return false or and error that is displayed to the user. (which these ones do now; generally speaking of all client -> server method calls)